### PR TITLE
Refactor PR

### PR DESCRIPTION
**Pull request by shreya-mishra**
Refactored file-  Mobile.js

### CODE REVIEW

**Review by lakshayman — APPROVED** [review]

File: app/controllers/mobile.js [hunk 1]
```diff
@@ -3,31 +3,35 @@ import { action } from '@ember/object';
...
-            method: 'PATCH',
-            headers: {
-              'Content-Type': 'application/json',
-            },
-            credentials: 'include',
-          }
-        );
-
-        if (response.status !== 200) {
+        const checkStatus = this.fetchAuthStatus(AUTH_STATUS.AUTHORIZED);
```
Comment (prakashchoudhary07) [inline app/controllers/mobile.js:33]: Should await be added here?

**Review by iamitprakash — APPROVED** [review]

File: app/constants/url.js [hunk 2]
```diff
@@ -9,3 +9,6 @@ export const MAIN_SITE_URL = 'https://realdevsquad.com/';
 export const GOTO_URL = 'https://realdevsquad.com/goto';
 export const AUTH_URL =
   'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97';
+export const FETCH_AUTH_STATUS =
+  'https://realdevsquad.com/auth/qr-code-auth/authorization_status/';
```
Comment (RitikJaiswal75) [inline app/constants/url.js:13]: I think we can use the env.baseApiUrl here

File: app/controllers/mobile.js [hunk 3]
```diff
@@ -3,31 +3,35 @@ import { action } from '@ember/object';
...
 import { toastNotificationTimeoutOptions } from '../constants/toast-notification';
+import { AUTH_STATUS } from '../constants/auth-status';
 
 export default class MobileController extends Controller {
   @service toast;
   @service router;
 
+  async fetchAuthStatus(authStatus) {
+    const response = await fetch(
+      `${ENV.BASE_API_URL}/auth/qr-code-auth/authorization_status/${authStatus}`,
```
Comment (RitikJaiswal75) [inline app/controllers/mobile.js:14]: this url can be relpaced with the constant created in app constants URLs

File: app/controllers/mobile.js [hunk 4]
```diff
@@ -41,21 +45,31 @@ export default class MobileController extends Controller {
...
-        `${ENV.BASE_API_URL}/auth/qr-code-auth-get-device-info?user_id=${this.model.userId}`,
-        {
-          method: 'GET',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-          credentials: 'include',
-        }
-      );
+      const response = await fetch(`${ENV.BASE_API_URL}/auth/device`, {
```
Comment (RitikJaiswal75) [inline app/controllers/mobile.js:66]: i think we can also create a constant for this URL so we can change all the URLs from a single place, managing the code as it grows will become easier.

**Review by sahsisunny — APPROVED** [review]

**Review by j24m — APPROVED** [review]
Changes look fine to me.

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**